### PR TITLE
Make VmManager more reliable in terms of catching runtime exceptions 

### PR DESCRIPTION
**Pull request by xnull**
## Overview

Description:
Make VmManager more reliable in terms of catching runtime exceptions from ViJava lib

Why should this be merged: 

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: it/src/main/java/org/corfudb/universe/universe/vm/VmManager.java [hunk 1]
```diff
@@ -284,16 +297,20 @@ private VirtualMachineCloneSpec createLinuxCustomization(ImmutableMap<String, St
...
-            return powerOn().map(taskInfo -> this);
+        try {
+            VirtualMachinePowerState powerState = vm.getSummary().runtime.powerState;
+            if (powerState == VirtualMachinePowerState.poweredOn) {
+                return Result.ok(this);
+            } else {
+                return powerOn().map(taskInfo -> this);
+            }
+        } catch (RuntimeException ex) {
+            return Result.error(new UniverseException(ex));
```
Comment (hisundar) [inline it/src/main/java/org/corfudb/universe/universe/vm/VmManager.java]: Do you want a message for this exception?

File: it/src/main/java/org/corfudb/universe/universe/vm/VmManager.java [hunk 2]
```diff
@@ -210,33 +218,38 @@ public IpAddress getResolvedIpAddress() {
...
 
-            return executeTask(() -> vmTemplate.cloneVM_Task(folder, vmName.getName(), cloneSpec));
+                    folder = new Folder(vmTemplate.getServerConnection(), folderR);
+                } else {
+                    folder = (Folder) vmTemplate.getParent();
+                }
+
+                return executeTask(() -> vmTemplate.cloneVM_Task(folder, vmName.getName(), cloneSpec));
+            } catch (RuntimeException ex) {
+                return Result.error(new UniverseException(ex));
```
Comment (hisundar) [inline it/src/main/java/org/corfudb/universe/universe/vm/VmManager.java]: Maybe a message for this exception

File: it/src/main/java/org/corfudb/universe/universe/vm/VmManager.java (hunk 1 above)
Comment (xnull) [reply it/src/main/java/org/corfudb/universe/universe/vm/VmManager.java]: not needed, it's completely clear

File: it/src/main/java/org/corfudb/universe/universe/vm/VmManager.java (hunk 2 above)
Comment (xnull) [reply it/src/main/java/org/corfudb/universe/universe/vm/VmManager.java]: not needed, it's completely clear

**Pull request merged**